dev/source/docs/ros2-over-ethernet.rst

```diff
@@ -0,0 +1,93 @@
+.. _ros2-over-ethernet:
+
+===================
+ROS 2 over Ethernet
+===================
+
+The following tutorial explains how to connect a flight controller to a companion computer using ethernet for DDS communication.
+
+Ensure you have ROS 2 :ref:`installed <ros2>` and have run :ref:`SITL <ros2-sitl>` successfully before attempting this page.
+
+Additionally, make sure you understand the :ref:`basics of networking in ArduPilot. <common-network>`
+
+==========
+Motivation
+==========
+
+Starting in ArduPilot 4.5, ethernet is now supported as a new communication interface.
+Compared to connecting an autopilot over serial, ethernet has the following advantages:
+
+* Ethernet is more immune to noise because it uses twisted pair wiring
+* Ethernet can run multiple protocols over the same protocol (unless you use PPP)
+* Ethernet is easier to debug with tools such as ``tcpdump`` or ``Wireshark``
+* Ethernet uses standard cabling, so it's more difficult to mix up TX and RX
+
+===============
+Physical Equipment
+===============
+
+The following equipment is required to complete this tutorial:
+
+* A computer that can run the MicroROS Agent and the ROS 2 CLI
+* An autopilot with ethernet support
+
+=============================================
+Necessary parameters for static configuration
+=============================================
+
+This list of parameters is given as an example.
+If you had the following static IP addresses:
+
+* An autopilot has the IP address ``192.168.1.6``
+* A Computer running the MicroROS agent has the IP address ``192.168.1.5``
+* The MicroROS agent is running on port ``2019``
+
+Then, you would configure all of the below parameters.
+
+
+.. list-table::
+   :widths: 50 50
+   :header-rows: 1
+
+   * - Parameter Name
+     - Value
+   * - DDS_ENABLE
+     - 1
+   * - NET_ENABLED
+     - 1
+   * - NET_DHCP
+     - 0
+   * - NET_IP_ADDR0
+     - 192
+   * - NET_IP_ADDR1
+     - 168
+   * - NET_IP_ADDR2
+     - 1
+   * - NET_IP_ADDR3
+     - 5
+   * - DDS_IP_0
```

Comment:
Why is this DDS_IP_0 when 1,2,3 below use NET ... NET_IP_1 ...
This really doesn't make sense. Why is this not a NET_P1_PROTOCOL?

Reply:
NET_Px is only for PPP Serial. This wiki is for dedicated ethernet (on a 6X or CubeRed)

Reply:
> NET_Px is only for PPP Serial. This wiki is for dedicated ethernet (on a 6X or CubeRed)

Definitely not true. I've set this up on a Cube Red - you need to specify the protocol and port numbers for mavlink over UDP and TCP. DDS should work the same.

Reply:
Feel free to give it a go. Here's my params, nothing is set for PPP and it works.
```
MANUAL> NET_DHCP         0           # Disable
NET_ENABLED      1           # Enable
NET_GWADDR0      192
NET_GWADDR1      168
NET_GWADDR2      1
NET_GWADDR3      1
NET_IPADDR0      192
NET_IPADDR1      168
NET_IPADDR2      1
NET_IPADDR3      6
NET_MACADDR0     194
NET_MACADDR1     175
NET_MACADDR2     81
NET_MACADDR3     213
NET_MACADDR4     226
NET_MACADDR5     14
NET_NETMASK      24
NET_OPTIONS      0
NET_P1_TYPE      0           # Disabled
NET_P2_TYPE      0           # Disabled
NET_P3_TYPE      0           # Disabled
NET_P4_TYPE      0           # Disabled
```

Reply:
Well I don't think it should be different, but if it is and if these are your parameters, where is the DDS_IP_0 that I first commented on?

Reply:
That was a dump with ` param show NET*`. Here's the same for DDS:
```
param show DDS*
MANUAL> DDS_ENABLE       1           # Enabled
DDS_IP0          192
DDS_IP1          168
DDS_IP2          1
DDS_IP3          5
DDS_PORT         2019
```

Reply:
What would you change, and what would you expect it to be configured like?

Reply:
I'd want to see it basically match the way it works for mavlink (e.g. using P2 assuming P1 is used by mavlink)
NET_P2_TYPE = 0  (UDP client)
NET_P2_IP0 = 192
NET_P2_IP1 = 168
NET_P2_IP2 = 1
NET_P2_IP3 = 5
NET_P2_PORT = 2019
NET_P2_PROTOCOL = 47 (new value for DDS)

Reply:
Ok. The approach I document here is different and is only documenting the outcome of last year's GSOC project. Configuring DDS to use PPP is possible, but it's not the point of the PR.

Reply:
Yes I see that now, it seems the DDS code pre-dates the NET stuff, but its not about PPP, what I'm talking about applies for all IP based networking, including with onboard hardware ethernet like the Cube Red or Pixhawk 6X.